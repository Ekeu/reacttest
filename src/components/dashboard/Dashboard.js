import React, { useEffect, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Spinner from '../layout/Spinner';
import BandWidth from './BandWidth';
import Concurrent from './Concurrent';
import { extractCurrentClient } from '../../actions/profile';
import { extractchartData } from '../../actions/chart';
import { extractConcurrentViewers } from '../../actions/viewers';

const Dashboard = ({
  extractCurrentClient,
  extractchartData,
  extractConcurrentViewers,
  auth: { user },
  profile: { profile, loading },
  chart: { cdn, p2p, maxSum, maxCdn },
  viewers: {concurrence}
}) => {
  useEffect(() => {
    extractCurrentClient();
    extractchartData('max', 1585834831000, 1589118031000);
    extractConcurrentViewers(1585834831000, 1589118031000);
  }, []);
  return loading && profile === null ? (
    <Spinner />
  ) : (
    <Fragment>
      <h1 className='large text-primary'>Streaming</h1>
      <p className='lead'>
        <i className='fas fa-chart-line'></i>
        Welcome {user && user.lname}
      </p>
      <BandWidth cdn={cdn} p2p={p2p} maxSum={maxSum} maxCdn={maxCdn} />
      <Concurrent concurrence={concurrence}/>
    </Fragment>
  );
};

Dashboard.propTypes = {
  extractCurrentClient: PropTypes.func.isRequired,
  extractchartData: PropTypes.func.isRequired,
  extractConcurrentViewers: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  profile: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  auth: state.auth,
  profile: state.profile,
  chart: state.chart,
  viewers: state.viewers,
});

export default connect(mapStateToProps, {
  extractCurrentClient,
  extractchartData,
  extractConcurrentViewers
})(Dashboard);
