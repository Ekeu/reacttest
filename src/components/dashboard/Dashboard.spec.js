import React from 'react';
import { mount, toJson, shallow } from '../../../tools/config/test/enzyme';
import Dashboard from './Dashboard';

let [
  extractCurrentClient,
  extractchartData,
  extractConcurrentViewers,
] = new Array(3).fill(jest.fn());

function setup() {
  const props = {
    auth: {
      user: {
        lname: 'Ulrich',
      },
    },
    profile: {
      profile: {
        fname: 'Ulrich',
        lname: 'Ekeu',
        email: 'hello@gmail',
        website: 'website.com',
        description: 'Hello Ulrich',
      },
      loading: true,
    },
    chart: {
      cdn: [
        [1234, 123],
        [1234, 123],
        [1234, 123],
      ],
      p2p: [
        [1234, 123],
        [1234, 123],
        [1234, 123],
      ],
      maxSum: 345.5,
      maxCdn: 345.5,
    },
    viewers: {
        concurrence: [
            [1234, 123],
            [1234, 123],
            [1234, 123],
          ]
    },
    extractCurrentClient: extractCurrentClient,
    extractchartData: extractchartData,
    extractConcurrentViewers: extractConcurrentViewers,
  };

  const enzymeWrapper = shallow(<Dashboard {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}
jest.mock('react-redux', () => ({ connect: () => (c) => c }));
jest.mock('../../img/spinner.gif', () => 'mock-bandwith');
const { enzymeWrapper } = setup();

describe('Dashboard Component', () => {
  it('should render correctly without props', () => {
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  });
});
