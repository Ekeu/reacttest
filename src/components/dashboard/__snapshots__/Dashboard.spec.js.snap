// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should render correctly without props 1`] = `
<Fragment>
  <h1
    className="large text-primary"
  >
    Streaming
  </h1>
  <p
    className="lead"
  >
    <i
      className="fas fa-chart-line"
    />
    Welcome 
    Ulrich
  </p>
  <DashBoardActions />
  <BandWidth
    cdn={
      Array [
        Array [
          1234,
          123,
        ],
        Array [
          1234,
          123,
        ],
        Array [
          1234,
          123,
        ],
      ]
    }
    maxCdn={345.5}
    maxSum={345.5}
    p2p={
      Array [
        Array [
          1234,
          123,
        ],
        Array [
          1234,
          123,
        ],
        Array [
          1234,
          123,
        ],
      ]
    }
  />
  <Concurrent
    concurrence={
      Array [
        Array [
          1234,
          123,
        ],
        Array [
          1234,
          123,
        ],
        Array [
          1234,
          123,
        ],
      ]
    }
  />
</Fragment>
`;
